tachyon/math/polynomials/univariate/lagrange_interpolation.h

```diff
@@ -0,0 +1,85 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_MATH_POLYNOMIALS_UNIVARIATE_LAGRANGE_INTERPOLATION_H_
+#define TACHYON_MATH_POLYNOMIALS_UNIVARIATE_LAGRANGE_INTERPOLATION_H_
+
+#include <utility>
+#include <vector>
+
+#include "tachyon/base/containers/container_util.h"
+#include "tachyon/base/parallelize.h"
+#include "tachyon/math/polynomials/univariate/univariate_polynomial.h"
+
+namespace tachyon::math {
+
+template <size_t MaxDegree, typename Container>
+bool LagrangeInterpolate(
+    const Container& points, const Container& evals,
+    UnivariateDensePolynomial<typename Container::value_type, MaxDegree>* ret) {
+  using F = typename Container::value_type;
+  using Poly = UnivariateDensePolynomial<F, MaxDegree>;
+  using Coeffs = UnivariateDenseCoefficients<F, MaxDegree>;
+
+  if (points.size() != evals.size() || points.size() != MaxDegree + 1) {
+    return false;
```

Comment:
Could you just logging why it's failed for user? 

```suggestion
    LOG(ERROR) << "points and evals size don't match";
    return false;
```

Reply:
`points.size() != MaxDegree + 1` this comparison looks redundant!